src/main/scala/jigg/util/JSONUtil.scala

```diff
@@ -18,18 +18,55 @@ object JSONUtil {
 
   private def toJSONFromNode(node: Node): String = {
     val sb = new StringBuilder
-
+    val escapedsb = new StringBuilder
+    val returnsb = new StringBuilder
     sb.append('{')
     sb.append(List("\".tag\":\"",node.label,"\",").mkString)
     sb.append("\".child\":")
     sb.append("[")
-    
     sb.append(serializing(node))
-
     sb.append("]")
     sb.append("}")
+    val returnString = pretty(render(parse(escapeString(sb.toString,escapedsb).toString)))
+    returnString
+  }
+
+  /**
+    * Append escaped string as 's'.
+    * This method is borrowed from xml.Utility.escape
+    */
+  final def escapeString(text: String, s: StringBuilder): StringBuilder = {
+    val length = text.length
+    var pos = 0
+    while (pos < length){
+      text.charAt(pos) match {
+        case '\\' => s.append("\\\\")
+        case c => if (c >= ' ') s.append(c)
+      }
+      pos += 1
+    }
+    s
+  }
 
-    pretty(render(parse(sb.toString)))
+  final def unEscapeString(text: String, s: StringBuilder): StringBuilder = {
+    val length = text.length
+    val backslash = "\\\\"
+    var pos = 0
+    while (pos < length){
+      val head = text.substring(pos, length).indexOf(backslash)
```

Comment:
Maybe the same thing can be done with `indexOf(backslash, pos)` which does not perform `substring`?